ksqldb-functional-tests/src/test/resources/query-validation-tests/comparison-expression.json

```
@@ -0,0 +1,74 @@
+{
+  "comments": [
+    "Tests covering comparison expressions"
+  ],
+  "tests": [
+    {
+      "name": "invalid where predicate - type mismatch",
+      "statements": [
+        "CREATE STREAM INPUT (ROWKEY DOUBLE KEY, ID INT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM INPUT WHERE ID >= 'not an int' EMIT CHANGES;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
```

We need  ALL of these negative tests to check for `KsqlStatementException`, as this then references the exact expression with the issue.

It's possible the right exception is being thrown already - just update the test and re-run.

```suggestion
        "type": "io.confluent.ksql.util.KsqlStatementException",
```

Note: in the validation code you throw a `KsqlException`. Higher up the call stack, (where the code had the full statement), this should be getting caught and converted to a `KsqlStatementException`.  There is no need to pass down the SQL so you can throw `KsqlStatementException` from lower down.